### DISCUSSION TITLE
Post upload file res.onData after async

### DISCUSSION THREAD

**Discussion by jeffery30162** [uNetworking/uWebSockets.js#1051] [Q&A] [answered]
Hello, I am having difficulties with receiving all of a files data when using async method before onData.
If I remove the async functionality all works fine. With it, the file ends up missing a few thousand bytes.

Goal:
I'm using nodejs workers, I need to send another worker a message and wait for the response before allowing the upload.

Issue:
When using res.onData after async, not all data chunks are captured.

Code Example:

      this.httpServer.post('/upload', (res, req) => {
		
	        res.onAborted(() => {
                res.aborted = true;
            });
			
            this.sendMasterMessage('verify_user', { user_id: user_id, type: 'upload' }).then((allowed) => {
			
			if (res.aborted) {
			    return;
			}

			if (!allowed) { return this.send404Response(res); }
			
			fileStream = fs.createWriteStream(
                           path.resolve(file_path)
                    );
				
			res.onData((chunk, isLast) => {
				fileStream.write(Buffer.concat([Buffer.from(chunk)]));
				if (isLast) {
					fileStream.end();
				}
			});
				
		});
			
	});

**Comment by uNetworkingAB** [comment, marked as answer]
See https://github.com/uNetworking/uWebSockets.js/discussions/1043
